### ISSUE TITLE
11909 Bus error on Gatsby build.

### ISSUE THREAD

**Issue by emilpriver** [gatsbyjs/gatsby#35049]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I get "11909 Bus error" when building a gatsby site. It happened recently when building. Did work fine before. 
I don't get the error locally but get it on Cloudflare Pages, Gatsby Cloud, and Netlify.
I am not able to create a minimal repo for this.

### Reproduction Link

https://github.com/emilpriver/gatsby-markdown-portfolio

### Steps to Reproduce

1. Deploy to Gatsby cloud, Cloudflare PAges. First deploys look to work but the other dont.
2. re-build projects
3.
...

### Expected Result

Succesfull build

### Actual Result

I get this error.

22:27:32 PM:
/usr/local/bin/runner-command: line 33: 11909 Bus error               (core dumped) ${GATSBY} build --no-color

### Environment

```shell
System:
    OS: macOS 12.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.7.0 - ~/.nvm/versions/node/v16.7.0/bin/node
    Yarn: 1.22.17 - ~/.yarn/bin/yarn
    npm: 7.20.3 - ~/.nvm/versions/node/v16.7.0/bin/npm
  Languages:
    Python: 3.9.9 - /usr/local/opt/python3/libexec/bin/python
  Browsers:
    Chrome: 99.0.4844.51
    Firefox: 86.0.1
    Safari: 15.2
  npmPackages:
    gatsby: ^4.6.0 => 4.9.1 
    gatsby-plugin-breadcrumb: ^12.3.1 => 12.3.1 
    gatsby-plugin-feed: ^4.6.0 => 4.9.0 
    gatsby-plugin-gatsby-cloud: ^4.6.0 => 4.9.0 
    gatsby-plugin-google-analytics: ^4.6.0 => 4.9.0 
    gatsby-plugin-image: ^2.6.0 => 2.9.0 
    gatsby-plugin-manifest: ^4.6.0 => 4.9.0 
    gatsby-plugin-offline: ^5.6.0 => 5.9.0 
    gatsby-plugin-postcss: ^5.6.0 => 5.9.0 
    gatsby-plugin-react-helmet: ^5.6.0 => 5.9.0 
    gatsby-plugin-sharp: ^4.6.0 => 4.9.0 
    gatsby-plugin-sitemap: ^5.6.0 => 5.9.0 
    gatsby-remark-autolink-headers: ^5.6.0 => 5.9.0 
    gatsby-remark-copy-linked-files: ^5.6.0 => 5.9.0 
    gatsby-remark-images: ^6.6.0 => 6.9.0 
    gatsby-remark-prismjs: ^6.6.0 => 6.9.0 
    gatsby-remark-reading-time: ^1.1.0 => 1.1.0 
    gatsby-remark-responsive-iframe: ^5.6.0 => 5.9.0 
    gatsby-remark-smartypants: ^5.6.0 => 5.9.0 
    gatsby-source-custom-api: ^2.3.2 => 2.3.2 
    gatsby-source-filesystem: ^4.6.0 => 4.9.0 
    gatsby-source-instagram-all: ^5.2.1 => 5.2.1 
    gatsby-source-spotify: ^2.0.0 => 2.0.0 
    gatsby-source-unsplash: ^1.0.3 => 1.0.3 
    gatsby-transformer-remark: ^5.6.0 => 5.9.0 
    gatsby-transformer-sharp: ^4.6.0 => 4.9.0
```

**Comment by mahtd** [comment]
If it's the same issue I was having, it should be fixed by upgrading lmdb to v2.2.3 https://github.com/DoctorEvidence/lmdb-js/issues/142#issuecomment-1051262586

**Comment by emilpriver** [comment]
> We're seeing the same on our 4.9 branch builds on Gatsby Cloud.
> 
> ```
> 22:12:42 PM:
> 
> /usr/local/bin/runner-command: line 33:  7749 Bus error               (core dumped) ${GATSBY} build --no-color
> 22:12:42 PM:
> 
> ERROR Failed to compile: Error: Exited with code 135
> ```

I got it locally 1 time but have got it multiple times when building in CI/CD. I tried with a gatsby clean before building which seems to work. Add `gatsby clean` to `preBuild` command

**Comment by smastrom** [comment]
I also had the same issue once I updated Gatsby (and related plugins) to latest version (4.9). This error occurs only when building.

**Errors**
_Gatsby Cloud:_
An error occurred during parallel query running.
_Terminal:_
runner-command: line 33: 23137 Bus error

I had to revert Gatsby to 4.5 in order to have everything working as usual. Not sure if this can help, in any case this is my current configuration:

**package.json**
```json
 ...
  "dependencies": {
    "gatsby": "<=4.5.3",
    "gatsby-plugin-algolia": "<=0.23.1",
    "gatsby-plugin-facebook-pixel": "<=1.0.8",
    "gatsby-plugin-google-analytics": "<=4.5.0",
    "gatsby-plugin-image": "<=2.5.2",
    "gatsby-plugin-manifest": "<=4.5.2",
    "gatsby-plugin-react-helmet": "<=5.5.0",
    "gatsby-plugin-sharp": "<=4.5.2",
    "gatsby-plugin-sitemap": "<=5.5.0",
    "gatsby-plugin-styled-components": "<=5.5.0",
    "gatsby-source-datocms": "<=3.0.12",
    "gatsby-source-filesystem": "<=4.5.2",
    "gatsby-transformer-remark": "<=5.5.2",
    "gatsby-transformer-sharp": "<=4.5.0",
    ...
```

**Comment by wardpeet** [comment]
Is it possible to give use the `lmdb` version by running.
```
npm ls lmdb
yarn why lmdb
```

**Comment by LekoArts** [comment]
This should be fixed in `gatsby@4.9.2`
